libnmstate/nm/nmclient.py

```diff
@@ -49,6 +51,13 @@ def nm_version():
     return NM.Client.get_version(client())
 
 
+def libnm_nm_version_mismatch_check():
+    if StrictVersion(NM.utils_version()) < StrictVersion(
```

Comment:
I think this should check for inequality, we want to warn if they are not equal in this case.

Reply:
I re-read the issue. Unfortunately it was not very clear, I am sorry for the confusion. There are actually three warnings

1) Warn if the versions do not match
2) Warn if the NM version is too old, currently we require at least 1.22.8 (the version should be a constant, probably in `libnmstate/__init__.py` with a name like `_MINIMUM_NM_VERSION`. @ffmancera what do you think?
3) (optional, not sure if it also makes sense) Warn if the utils version is too old

I hope things are more clear now. What do you think?

Reply:
Yes, things are clearer now. I'll implement the first two warnings. For the 3rd warning what should the lower limit of the utils_version be?

Reply:
I have another query about the warning with comparison between the versions. Are the version numbers of libnm and nm supposed to be equal for sure for them to be compatible? Is there something like the versions are compatible even if libnm version is greater to nm version? Should the warning be given even for the compatible versions of libnm and nm?

Reply:
> Yes, things are clearer now. I'll implement the first two warnings. For the 3rd warning what should the lower limit of the utils_version be?

the same version

> I have another query about the warning with comparison between the versions. Are the version numbers of libnm and nm supposed to be equal for sure for them to be compatible? Is there something like the versions are compatible even if libnm version is greater to nm version? Should the warning be given even for the compatible versions of libnm and nm?

Good question. For now, please warn if it is not an exact match. 1.22.x is probably compatible with 1.22.y but we do not expect anyone to do want this intentionally, therefore I favor the exact match